### PR TITLE
Change the visual of a game

### PR DESCRIPTION
**Pull request by arimet**
- Modification of the marbles for a more real rendering 
- Changing the board

https://user-images.githubusercontent.com/18294269/135429245-431899b6-b400-4ce5-8480-e9fbc8baad4b.MOV

### CODE REVIEW

File: src/GameState/Marble.tsx [hunk 1]
```diff
@@ -1,12 +1,25 @@
-import { Circle, Pressable, Box } from 'native-base';
+import { Circle, Pressable, Box, View } from 'native-base';
 import React from 'react';
+import { ImageBackground } from 'react-native';
 
-const MARBLE_COLOR_RADIAL = ['warmGray.300', 'black', 'white', 'red.700'];
+const RED_MARBLE = require('./red-marble.png');
+const WHITE_MARBLE = require('./white-marble.png');
+const BLACK_MARBLE = require('./black-marble.png');
```
Comment (Luwangel) [inline src/GameState/Marble.tsx:7]: The import keyword doesn't work?

File: src/GameState/Marble.tsx [hunk 2]
```diff
@@ -1,12 +1,25 @@
-import { Circle, Pressable, Box } from 'native-base';
+import { Circle, Pressable, Box, View } from 'native-base';
 import React from 'react';
+import { ImageBackground } from 'react-native';
 
-const MARBLE_COLOR_RADIAL = ['warmGray.300', 'black', 'white', 'red.700'];
+const RED_MARBLE = require('./red-marble.png');
```
Comment (guilbill) [inline src/GameState/Marble.tsx:5]: It's ok, but you could have use an svg, and change the color, resulting in loading less ressources. But again, it's ok

File: src/GameState/Marble.tsx [hunk 3]
```diff
@@ -1,12 +1,25 @@
...
+const WHITE_MARBLE = require('./white-marble.png');
+const BLACK_MARBLE = require('./black-marble.png');
+const MARBLE_COLOR_IMAGE = ['', BLACK_MARBLE, WHITE_MARBLE, RED_MARBLE];
```
Comment (guilbill) [inline src/GameState/Marble.tsx]: Why is there an empty string in the array?

File: src/GameState/index.tsx [hunk 4]
```diff
@@ -112,9 +111,21 @@ export default function GameState({ navigation, route, player }: any) {
...
+
+  const getOpponentUserInformation = () => {
+    const opponentUserInformation = state.players.find(
+      (p) => p.id !== player.id,
+    );
+    return getPlayerObject(opponentUserInformation);
+  };
+
+  const getPlayerObject = (player: User | undefined) => {
     const playerGame = state?.game?.players?.find(
```
Comment (Luwangel) [inline src/GameState/index.tsx:125]: state and dispatch are really bad names...

**Review by Luwangel — COMMENTED** [review]
It would have been better with svg images.

File: src/GameState/index.tsx [hunk 5]
```diff
@@ -112,9 +111,21 @@ export default function GameState({ navigation, route, player }: any) {
     }
   };
 
-  const getPlayerObject = (player: User) => {
+  const getMobileUserInformation = () => {
```
Comment (guilbill) [inline src/GameState/index.tsx]: You could have one method, passing the user id as param. Also, I found the function name confusing, are we getting information specific for mobile? I would go for a `getUserInformation(userId)`.

**Review by guilbill — CHANGES_REQUESTED** [review]
The GameState/index.tsx is growing, you may split it in different components. I personnaly prefer to have a file by component, and the index.ts export those components.

Also, try to be consistent in the way you declare function, sometimes it's `function () {...}` and sometimes `()=>{...}`, stick to one.

**Review by guilbill — APPROVED** [review]

**Pull request merged**